### PR TITLE
Pr/use foundation hooks

### PR DESCRIPTION
**Pull request by sbabcoc**

### CODE REVIEW

**Comment by sbabcoc** [comment]
`ReportPortalListener` only has one conflict... the entire file! ;-)

**Comment by sbabcoc** [comment]
Given that the entire content of `ReportPortalListener` has been replaced, perhaps it would be better to simply delete this file and replace it with a new file named `ReportPortalWatcher`.

**Comment by sbabcoc** [comment]
Fixes a bug in JUnit Foundation for ignored tests

File: src/test/resources/reportportal.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-rp.endpoint=http://10.11.211.101:8080
-rp.uuid=452eab16-e69b-49cc-a482-836864d9f11c
-rp.launch=scott.babcock_JUnit_Agent
-rp.project=rfds
+rp.endpoint=http://reportportal.mpe.nordstrom.com:8080
```
[the remark this answers is not in the document]
Comment (sbabcoc) [reply src/test/resources/reportportal.properties]: Replaced with generic example

**Comment by DzmitryHumianiuk** [comment]
@sbabcoc hi, we finally approaching to review your changes.
approach looks promising and it's a right way to go with jUnit4.

We discussed it with @avarabyeu and decided to give it a try first.
We will merge it to separate branch, `v4`, create pull request to merge it into `master`.
With Jitpack.io create dependency for that particular revision, and ask users to try it with jUnit4.
This will help us to make public testing, and fix possible issue before publishing.

**Comment by DzmitryHumianiuk** [comment]
@sbabcoc could you plz resolve conflicts?

**Comment by sbabcoc** [comment]
Yes, with the scope of changes, a beta release seems like a good idea.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by sbabcoc** [comment]
Resolved merge conflicts

**Pull request merged**